### ISSUE TITLE
libharu can fail to install due to unmet dependency

### ISSUE THREAD

**Issue by scottwittenburg** [spack/spack#6681]
On my linux machine where I recently tried this, I think it worked because the configure step found `zlib` and `png` in `/usr`.  However, the `libharu` spack package should probably make those dependencies explicit.

When I tried to install `libharu` on my macbook today, it failed with the following error output in the terminal:

```console
$ spack -d install libharu
==> '/usr/bin/git' 'ls-tree' '--name-only' 'HEAD'
==> Reading config file /Users/scott/projects/spack/etc/spack/defaults/packages.yaml
==> Reading config file /Users/scott/projects/spack/etc/spack/defaults/darwin/packages.yaml
==> READ LOCK: /Users/scott/.spack/cache/providers/.builtin-index.yaml.lock[0:0] [Acquiring]
==> READ LOCK: /Users/scott/.spack/cache/providers/.builtin-index.yaml.lock[0:0] [Acquired]
==> READ LOCK: /Users/scott/.spack/cache/providers/.builtin-index.yaml.lock[0:0] [Released]
==> Reading config file /Users/scott/.spack/darwin/compilers.yaml
==> WRITE LOCK: /Users/scott/projects/spack/opt/spack/.spack-db/prefix_lock[6176013292513912332:1] [Acquiring]
==> WRITE LOCK: /Users/scott/projects/spack/opt/spack/.spack-db/prefix_lock[6176013292513912332:1] [Acquired]
==> READ LOCK: /Users/scott/projects/spack/opt/spack/.spack-db/lock[0:0] [Acquiring]
==> READ LOCK: /Users/scott/projects/spack/opt/spack/.spack-db/lock[0:0] [Acquired]
==> READ LOCK: /Users/scott/projects/spack/opt/spack/.spack-db/lock[0:0] [Released]
==> WRITE LOCK: /Users/scott/projects/spack/opt/spack/.spack-db/prefix_lock[6176013292513912332:1] [Released]
==> READ LOCK: /Users/scott/projects/spack/opt/spack/.spack-db/prefix_lock[6176013292513912332:1] [Acquiring]
==> READ LOCK: /Users/scott/projects/spack/opt/spack/.spack-db/prefix_lock[6176013292513912332:1] [Acquired]
==> READ LOCK: /Users/scott/projects/spack/opt/spack/.spack-db/prefix_lock[6176013292513912332:1] [Released]
==> Installing libharu dependencies
==> Installing libharu
==> link /Users/scott/projects/spack/var/spack/stage/libharu-2.3.0-vnvtnpbyypgbsvu4dbdnst2gyey25kgd -> /tmp/scott/spack-stage/spack-stage-cKhVhX
==> Using cached archive: /Users/scott/projects/spack/var/spack/cache/libharu/libharu-2.3.0.tar.gz
==> Staging archive: /Users/scott/projects/spack/var/spack/stage/libharu-2.3.0-vnvtnpbyypgbsvu4dbdnst2gyey25kgd/RELEASE_2_3_0.tar.gz
==> '/usr/bin/tar' '-xf' '/Users/scott/projects/spack/var/spack/stage/libharu-2.3.0-vnvtnpbyypgbsvu4dbdnst2gyey25kgd/RELEASE_2_3_0.tar.gz'
==> Created stage in /Users/scott/projects/spack/var/spack/stage/libharu-2.3.0-vnvtnpbyypgbsvu4dbdnst2gyey25kgd
==> No patches needed for libharu
==> Building libharu [AutotoolsPackage]
==> WRITE LOCK: /Users/scott/projects/spack/var/spack/stage/.lock[8944532592762121411:1] [Acquiring]
==> WRITE LOCK: /Users/scott/projects/spack/var/spack/stage/.lock[8944532592762121411:1] [Acquired]
==> WRITE LOCK: /Users/scott/projects/spack/opt/spack/.spack-db/prefix_lock[6176013292513912332:1] [Acquiring]
==> WRITE LOCK: /Users/scott/projects/spack/opt/spack/.spack-db/prefix_lock[6176013292513912332:1] [Acquired]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> WRITE LOCK: /Users/scott/projects/spack/opt/spack/.spack-db/prefix_lock[6176013292513912332:1] [Released]
==> WRITE LOCK: /Users/scott/projects/spack/var/spack/stage/.lock[8944532592762121411:1] [Released]
==> Error: ProcessError: Command exited with status 1:
    '/Users/scott/projects/spack/var/spack/stage/libharu-2.3.0-vnvtnpbyypgbsvu4dbdnst2gyey25kgd/libharu-RELEASE_2_3_0/configure' '--prefix=/Users/scott/projects/spack/opt/spack/darwin-highsierra-x86_64/clang-9.0.0-apple/libharu-2.3.0-vnvtnpbyypgbsvu4dbdnst2gyey25kgd'

1 error found in build log:
     [ ... ]
     100   checking for unistd.h... (cached) yes
     101   checking for stdint.h... (cached) yes
     102   checking for floor in -lm... yes
     103   Updated include/hpdf_version.h
     104   checking Zlib install prefix... /usr
     105   checking for deflate in -lz... yes
  >> 106   checking libpng install prefix... configure: error: Unable to locate libpng headers, please use --with-png=<DIR>

```

### Information on your system

 This happened on Mac OS X High Sierra

**Comment by scottwittenburg** [comment]
I have pushed a possible fix: #6682

**Comment by adamjstewart** [comment]
Closing now that #6682 has been merged.
